### PR TITLE
Mouse event handlers parameters normalization

### PR DESCRIPTION
**Pull request by martinmanzo**
## Description
I think that all mouse event handlers should expose the same parameters no enable greater customization when handling mouse events (mouse over, mouse out and mouse move.

## Motivation and Context
While developing the Angular wrapper I encountered that the bar events aren't consistent and some even doesn't fire any information the interaction.

This pull request includes changes only to bar charts event handlers. If you like the change and the style I would make the changes to the rest of the charts.

## How Has This Been Tested?
I followed TDD. Write failing tests, write code, pass tests, refactor code and pass tests.

## Types of changes

- [ ] Bug fix (non-breaking change which fixes an issue)
- [X] New feature (non-breaking change which adds functionality)
- [ ] Breaking change (fix or feature that would cause existing functionality to change)

## Checklist:

- [X] My code follows the code style of this project.
- [ ] My change requires a change to the documentation.
- [ ] I have updated the documentation accordingly.
- [X] I have read the **CONTRIBUTING** document.
- [X] I have added tests to cover my changes.
- [X] All new and existing tests passed.

### CODE REVIEW

File: src/charts/bar.js [hunk 1]
```diff
@@ -597,6 +589,34 @@ define(function(require) {
                     .attr('y2', chartHeight);
         }
 
+        /**
+         * Custom OnMouseOver event handler
+         * @return {void}
+         * @private
+         */
+        function customOnMouseOver(ev, d, chartWidth, chartHeight) {
```
Comment (Golodhros) [inline src/charts/bar.js]: Let's call it 'e' instead of 'ev', thanks!

File: src/charts/bar.js [hunk 2]
```diff
@@ -597,6 +589,34 @@ define(function(require) {
...
+          dispatcher.call('customMouseOver', ev, d, d3Selection.mouse(ev), [chartWidth, chartHeight]);
```
Comment (Golodhros) [inline src/charts/bar.js]: Nit: let's keep 4 spaces indentation

File: src/charts/bar.js (hunk 1 above)
Comment (martinmanzo) [inline src/charts/bar.js]: Sure!

File: src/charts/bar.js (hunk 2 above)
Comment (martinmanzo) [inline src/charts/bar.js]: My bad 😅

**Comment by martinmanzo** [comment]
Thanks for the feedback!

I'll work on the rest of the charts and let you know when it's done.

**Comment by Golodhros** [comment]
Thanks a lot @martinmanzo !!

File: src/charts/bar.js [hunk 3]
```diff
@@ -597,6 +589,34 @@ define(function(require) {
                     .attr('y2', chartHeight);
         }
 
+        /**
+         * Custom OnMouseOver event handler
+         * @return {void}
+         * @private
+         */
+        function customOnMouseOver(e, d, chartWidth, chartHeight) {
```
Comment (Golodhros) [inline src/charts/bar.js]: Could we rename these to 'handleMouseOver', 'handleMouseMove' and 'handleMouseOut' ?

File: test/specs/bar.spec.js [hunk 4]
```diff
@@ -235,14 +235,37 @@ define(['d3', 'bar', 'barChartDataBuilder'], function(d3, chart, dataBuilder) {
...
+
+            it('should trigger a callback on mouse out', () => {
+                let bar = containerFixture.selectAll('.bar:nth-child(1)');
+                let callbackSpy = jasmine.createSpy('callback');
+
+                barChart.on('customMouseOut', callbackSpy);
+                bar.dispatch('mouseout');
+
+                expect(callbackSpy.calls.count()).toBe(1);
+                expect(callbackSpy.calls.allArgs()[0].length).toBe(3);
```
Comment (Golodhros) [inline test/specs/bar.spec.js]: Thanks for the tests!!

File: src/charts/bar.js (hunk 3 above)
Comment (martinmanzo) [inline src/charts/bar.js]: Sure! No problem.

**Comment by Golodhros** [comment]
The release is almost ready, do you think we could include this?

Thanks!!

**Comment by martinmanzo** [comment]
Hi!
For the moment I've done the bar, donut and grouped bar charts.
The remaining are line, sparkline (?), stacked area, stacked bar and step charts, right?

**Comment by Golodhros** [comment]
I guess the Sparkline is not necessary for now.
I can tackle line and stacked area if you want.

**Comment by martinmanzo** [comment]
Sure, you don't need to ask :smile: 
I'll tackle stacked bar and step charts now.

I tried removing the chartWidth and chartHeigh parameters from the bar chart handlers but it broke the mini tooltip demo. Removing those parameters would introduce a breaking change.

**Comment by Golodhros** [comment]
Yeah, those are needed for now.
I was thinking on adding them always, but I am not super sure about it.

In general, we can think we have two tooltips that function in different ways. The 'Tooltip' is our older (and less consistent) one. The 'Mini-tooltip' I think is more clear code-wise.

There is a ticket for reworking the tooltips, and at that point I would love to merge them in one tooltip with different modes. It will be tricky. For this release though, I would be happy with having the tooltip APIs a bit more consistent, we shouldn't go crazy about it.

Thanks a lot!

**Comment by martinmanzo** [comment]
You are welcome!

I just finished stacked bar and step charts. Should I continue with line or stacked area or are you in that already?

**Comment by Golodhros** [comment]
Awesome!
If you could do it that would be great! I haven't started with this. 

I am trying to get everything ready today, just let me know if you need help with anything and I can tackle it later.

Thanks!

**Comment by martinmanzo** [comment]
Great! I'm on it. Should be ready today.

**Comment by martinmanzo** [comment]
Done! :tada:

File: src/charts/stacked-bar.js [hunk 5]
```diff
@@ -874,7 +880,11 @@ define(function(require){
 
         /**
          * Gets or Sets the number of ticks of the x axis on the chart
-         * (Default is 5)         
+<<<<<<< HEAD
```
Comment (Golodhros) [inline src/charts/stacked-bar.js]: Ooops, let's clean this

File: CHANGELOG.md [hunk 6]
```diff
@@ -1,3 +1,9 @@
+<<<<<<< HEAD
+<<<<<<< HEAD
+1.7.2
+=======
```
Comment (Golodhros) [inline CHANGELOG.md]: A merge conflict

File: test/specs/donut.spec.js [hunk 7]
```diff
@@ -194,6 +194,7 @@ define(['d3', 'donut', 'donutChartDataBuilder'], function(d3, chart, dataBuilder
                     donutChart.on('customMouseOver', callback);
                     firstSlice.dispatch('mouseover');
                     expect(callback.calls.count()).toBe(1);
+                    expect(callback.calls.allArgs()[0].length).toBe(3);
```
Comment (Golodhros) [inline test/specs/donut.spec.js]: Nice!

**Comment by martinmanzo** [comment]
Missed both of those details! Should be Ok now.

**Comment by Golodhros** [comment]
Awesome, thanks a lot!

**Pull request merged**